src/QMCWaveFunctions/Fermion/SlaterDet.cpp

```diff
@@ -39,15 +39,17 @@ SlaterDet::SlaterDet(ParticleSet& targetPtcl, const std::string& class_name) : W
 SlaterDet::~SlaterDet() = default;
 
 ///add a new DiracDeterminant to the list of determinants
-void SlaterDet::add(Determinant_t* det, int ispin)
+void SlaterDet::add(std::unique_ptr<Determinant_t> det, int ispin)
```

Comment:
This add member function makes it difficult to simply pass BackflowTransformation in the constructor. SlaterDetWithBackflow and MultiSlaterDeterminantWithBackflow (others??) would need to pass a pointer to BFTrans here or otherwise ensure that det is already pointing to the right object.